### PR TITLE
[WIP] Flink supports k8s operator submit task

### PR DESCRIPTION
**Pull request by ChengJie1053**
Flink supports k8s operator submit task

### CODE REVIEW

File: linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/deployment/FlinkDeploymentOperatorClusterDescriptorAdapter.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+          FlinkDeployment,
+          FlinkDeploymentList,
+          org.apache.flink.kubernetes.shaded.io.fabric8.kubernetes.client.dsl.Resource<
+              FlinkDeployment>>
+      getFlinkDeploymentClient() {
+    return client.customResources(FlinkDeployment.class, FlinkDeploymentList.class);
+  }
+
+  public FlinkDeployment getFlinkDeployment(String name, String namespace) {
+    FlinkDeployment FlinkDeployment = new FlinkDeployment();
```
Comment (rarexixi) [inline linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/deployment/FlinkDeploymentOperatorClusterDescriptorAdapter.java:209]: FlinkDeployment flinkDeployment = ...

File: linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/deployment/FlinkDeploymentOperatorClusterDescriptorAdapter.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    this.client =
+        FlinkKubernetesHelper.getKubernetesClientByKubeConfigFile(config.getK8sConfigFile());
+  }
+
+  public void deployCluster(String[] programArguments, String applicationClassName) {
+    logger.info("The flink k8s operator task start，k8sNamespace: {}", config.getK8sNamespace());
+    CustomResourceDefinitionList crds =
+        client.apiextensions().v1().customResourceDefinitions().list();
+
+    String FlinkDeploymentCRDName = CustomResource.getCRDName(FlinkDeployment.class);
```
Comment (rarexixi) [inline linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/deployment/FlinkDeploymentOperatorClusterDescriptorAdapter.java:71]: String flinkDeploymentCRDName = ...

File: linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/deployment/FlinkDeploymentOperatorClusterDescriptorAdapter.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            config.getFlinkAppName(),
+            jobStatus.toString());
+      }
+    }
+  }
+
+  public void startFlinkKubernetesOperatorWatcher() {
+    getFlinkDeploymentClient()
+        .inNamespace(this.config.getK8sNamespace())
+        .withName(this.config.getFlinkAppName())
```
Comment (rarexixi) [inline linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/deployment/FlinkDeploymentOperatorClusterDescriptorAdapter.java:155]: Consider replacing with label ticket-id

Comment (ChengJie1053) [reply linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/deployment/FlinkDeploymentOperatorClusterDescriptorAdapter.java:155]: Ok, thanks for reviewing the code

File: linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/deployment/FlinkDeploymentOperatorClusterDescriptorAdapter.java [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+              FlinkDeployment>>
+      getFlinkDeploymentClient() {
+    return client.customResources(FlinkDeployment.class, FlinkDeploymentList.class);
+  }
+
+  public FlinkDeployment getFlinkDeployment(String name, String namespace) {
+    FlinkDeployment FlinkDeployment = new FlinkDeployment();
+    ObjectMeta metadata = new ObjectMeta();
+    metadata.setName(name);
+    metadata.setNamespace(namespace);
```
Comment (rarexixi) [inline linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/deployment/FlinkDeploymentOperatorClusterDescriptorAdapter.java:212]: Set some labels here (like `"from-app": "linkis"`, `"linkis-ticket-id": ${ticketId}`)

File: linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/crds/diff/Diffable.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.linkis.engineconnplugin.flink.client.shims.crds.diff;
+
+/**
+ * {@link Diffable} classes can be compared with other {@link Diffable} objects for differences.
+ *
+ * <p>Inspired by:
+ * https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/builder/Diffable.java
```
Comment (peacewong) [inline linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/crds/diff/Diffable.java:24]: need to update

Comment (ChengJie1053) [reply linkis-engineconn-plugins/flink/flink-shims-1.16.2/src/main/java/org/apache/linkis/engineconnplugin/flink/client/shims/crds/diff/Diffable.java:24]: Ok, thanks for reviewing the code

**Pull request closed without merging**